File: analyzers/its/sources/ManuallyAddedNoncompliantIssuesVB/IntentionalFindingsVB/S2612.vb
Imports System.Security.AccessControl

Public Class S2612
    Public Sub TestCases(fileSecurity As FileSecurity, directorySecurity As DirectorySecurity)
        fileSecurity.SetAccessRule(new FileSystemAccessRule("User", FileSystemRights.ListDirectory, AccessControlType.Allow))

        fileSecurity.SetAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow)) ' Noncompliant (S2612) {{Make sure this permission is safe.}}

        directorySecurity.RemoveAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow))

        directorySecurity.AddAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow)) ' Noncompliant
    End Sub
End Class

File: analyzers/its/Projects/CSharpLatest/CSharpLatest/CSharp9Features/S2612.cs
using System.Security.AccessControl;

namespace CSharpLatest.CSharp9Features;

public class S2612
{
    void Foo(FileSecurity fileSecurity)
    {
        FileSystemAccessRule unsafeAccessRule = new ("Everyone", FileSystemRights.FullControl, AccessControlType.Allow);
        fileSecurity.AddAccessRule(unsafeAccessRule);
        fileSecurity.SetAccessRule(unsafeAccessRule);
    }
}
